web/service/inbound.go

```diff
@@ -1292,6 +1292,27 @@ func (s *InboundService) GetClientTrafficByEmail(email string) (traffic *xray.Cl
 	return nil, nil
 }
 
+func (s *InboundService) GetClientTrafficByID(id string) (traffic *xray.ClientTraffic,err error) {
+	db := database.GetDB()
+	var traffics []*xray.ClientTraffic
+	err = db.Model(xray.ClientTraffic{}).Where(`email IN(
+		SELECT JSON_EXTRACT(client.value, '$.email') as email
+		FROM inbounds,
+	  	JSON_EACH(JSON_EXTRACT(inbounds.settings, '$.clients')) AS client
+		WHERE
+	  	JSON_EXTRACT(client.value, '$.id') in (?)
+		)`, id).Find(&traffics).Error
+
+	if err != nil {
```

Comment:
return empty array instead of error in case of `gorm.ErrRecordNotFound` would be better

Reply:
It is still exists. I mean using this code instead:
```go
       if err != nil {
            logger.Debug(err)
	    return nil, err
        }
	return traffics, err
```

You don't need to taking care of empty answer